app/scripts/controllers/swaps.js

```diff
@@ -273,6 +258,11 @@ export default class SwapsController {
     isPolledRequest,
   ) {
     const { chainId } = fetchParamsMetaData;
+
+    if (chainId !== this._ethersProviderChainId) {
+      this.ethersProvider = new Web3Provider(this.provider);
```

Comment:
shouldn't `this._ethersProviderChainId` be updated to the new chainId?

Reply:
yes, thank-you